add-interactive.c

```diff
@@ -323,14 +574,19 @@ static struct choices *list_and_choose(struct choices *data,
 
 		for (i = 0; i < data->nr; i++) {
 			struct choice *c = data->choices[i];
+			char chosen = chosen_choices[i]? '*' : ' ';
 			const char *modified_fmt = _("%12s %12s %s");
 			char print_buf[100];
 
 			strbuf_reset(&print);
 			strbuf_reset(&index_changes);
 			strbuf_reset(&worktree_changes);
 
-			snprintf(print_buf, 100, "%s", c->name);
+			if (!opts->list_only)
+				snprintf(print_buf, 100, "%s",
+					 highlight_prefix(data->choices[i]));

```

Comment:
@dscho You suggested me to do the following for `highlight_prefix`: better pass that buffer in, as a function parameter, initializing it with `strbuf_reset(buf)`; first thing, and still returning `buf.buf` but as a `const char *.
I'm not sure I understand why and how... Did you mean to make brand new `strbuf` here and send it as parameter to `highlight_prefix`?

Reply:
I meant to replace `print_buf` by a `struct strbuf buf = STRBUF_INIT;`, and then pass that into `highlight_prefix(&buf, data->choices[i]);`